plugins/inputs/ping/ping.go

```diff
@@ -261,11 +269,14 @@ func (p *Ping) pingToURLNative(destination string, acc telegraf.Accumulator) {
 					Seq: seq,
 				})
 				if err != nil {
-					acc.AddFields("ping", map[string]interface{}{"result_code": 2}, map[string]string{"url": destination})
-					acc.AddError(err)
+					if !strings.Contains(err.Error(), "not permitted") {
+						packetsSent++
+					}
+					doErr = err
```

Comment:
Data race with doErr and packetsSent, likely want to package these up and send them into the channel.